src/main/java/io/cryostat/net/OpenShiftAuthManager.java

```diff
@@ -334,67 +369,104 @@ private String getTokenFromHttpHeader(String rawHttpHeader) {
-                                                                CRYOSTAT_OAUTH_CLIENT_ID)));
-
-                        String scope =
-                                String.format(
-                                        "user:check-access role:%s:%s",
-                                        roleScope.orElseThrow(
-                                                () ->
-                                                        new MissingEnvironmentVariableException(
-                                                                CRYOSTAT_OAUTH_ROLE)),
-                                        namespace);
+                        String oauthClient = this.getServiceAccountName();
+                        String tokenScope = this.getTokenScope();
 
-                        webClient
-                                .get(443, OAUTH_WELL_KNOWN_HOST, WELL_KNOWN_PATH)
-                                .putHeader("Accept", "application/json")
-                                .send(
-                                        ar -> {
-                                            if (ar.failed()) {
-                                                oauthMetadata.completeExceptionally(ar.cause());
-                                                return;
-                                            }
-                                            oauthMetadata.complete(ar.result().bodyAsJsonObject());
-                                        });
+                        CompletableFuture<JsonObject> oauthMetadata = this.computeOauthMetadata();
 
                         String authorizeEndpoint =
-                                oauthMetadata.get().getString(OAUTH_ENDPOINT_KEY);
+                                oauthMetadata.get().getString(AUTHORIZATION_URL_KEY);
 
                         URIBuilder builder = new URIBuilder(authorizeEndpoint);
-                        builder.addParameter("client_id", serviceAccountAsOAuthClient);
+                        builder.addParameter("client_id", oauthClient);
                         builder.addParameter("response_type", "token");
                         builder.addParameter("response_mode", "fragment");
-                        builder.addParameter("scope", scope);
+                        builder.addParameter("scope", tokenScope);
 
                         return CompletableFuture.completedFuture(builder.build().toString());
                     } catch (ExecutionException
                             | InterruptedException
                             | URISyntaxException
                             | IOException
                             | MissingEnvironmentVariableException e) {
-                        throw new CompletionException(e);
+                        return CompletableFuture.failedFuture(e);
                     }
                 });
     }
 
+    private CompletableFuture<String> computeLogoutRedirectEndpoint() {
```

Comment:
In order for the frontend to access the OAuth server logout endpoint, the APIServer needs the cryostat web host added to the spec `additionalCORSAllowedOrigins`. The [docs](https://docs.openshift.com/container-platform/4.8/rest_api/config_apis/apiserver-config-openshift-io-v1.html) mention this "may be needed to access the API and the integrated OAuth server from JavaScript applications". If I don't include this spec, the /logout request fails with a CORS error.
Here's a hardcoded example:
```
apiVersion: config.openshift.io/v1
kind: APIServer
metadata:
  ...
spec:
  additionalCORSAllowedOrigins:
  - cryostat-sample-default\.apps\.jalaw0\.lab\.upshift\.rdu2\.redhat\.com
``` 

The request would look something like this:
```
client.config()
                    .apiServers()
                    .edit(
                            APIServerSpec.class,
                            spec -> spec.setAdditionalCORSAllowedOrigins(cryostatOrigin));
```


Where's the best place to put this configuration request in the backend?

Reply:
IMO that fits best in -operator, actually. That configuration could be added when a Cryostat CR is created, and removed when the Cryostat CR is deleted.

Reply:
Sounds good, I created an issue for it